server/app/services/grid_service_deployer.rb

```diff
@@ -53,7 +53,12 @@ def deploy
     self.grid_service.grid_service_instances.where(:instance_number.gt => total_instances).destroy
     total_instances.times do |i|
       instance_number = i + 1
-      unless self.grid_service.reload.deploying?
+      self.grid_service_deploy.reload
+      if self.grid_service_deploy.abort?
+        raise "halting deploy of #{self.grid_service.to_path}, deploy was aborted: #{self.deploy_state.reason}"
+      end
+      self.grid_service.reload
+      unless self.grid_service.running? || self.grid_service.initialized?
```

Comment:
May this should just be `if self.grid_service.stopped?` - I think that's the only possible alternative state?